src/replay/service.zig

```diff
@@ -162,16 +510,30 @@ fn advanceReplay(state: *ReplayState) !void {
     try trackNewSlots(
         state.allocator,
         state.account_store,
-        &state.blockstore_db,
-        state.slot_tracker,
-        state.epochs,
+        &state.ledger.db,
+        &state.slot_tracker,
+        &state.epochs,
         state.slot_leaders,
-        state.execution.progress_map,
+        &state.progress,
     );
 
-    _ = try replay.execution.replayActiveSlots(&state.execution);
+    _ = try replay.execution.replayActiveSlots(state.executionState());
+
+    _ = try replay.edge_cases.processEdgeCases(state.allocator, state.logger, .{
+        .my_pubkey = state.my_identity,
+        .tpu_has_bank = false,
 
-    _ = &replay.edge_cases.processEdgeCases;
+        .fork_choice = &state.fork_choice,
+        .ledger = state.ledger.writer,
+
+        .slot_tracker = &state.slot_tracker,
+        .progress = &state.progress,
+        .latest_validator_votes = &state.latest_validator_votes,
+        .slot_data = &state.slot_data,
+
+        .senders = state.senders,
+        .receivers = state.receivers,
+    });
 
     try replay.consensus.processConsensus(null);
```

Comment:
Why is this null? Can you add the inputs here? Finishing processConsensus is the top priority for consensus.